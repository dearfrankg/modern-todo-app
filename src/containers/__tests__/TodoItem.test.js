import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import TodoItemContainer, { mapDispatchToProps } from 'containers/TodoItem';

const middlewares = [];
const mockStore = configureStore(middlewares);

const state = {};
const store = mockStore(state);

describe('TodoItem Container:', () => {
  it('should render correctly with default props', () => {
    expect(
      shallow(
        <Provider store={store}>
          <TodoItemContainer />
        </Provider>,
      ),
    ).toMatchSnapshot();
  });

  describe('should call actions', () => {
    const dispatchSpy = jest.fn();
    const { toggleTodo, deleteTodo, editTodo } = mapDispatchToProps(dispatchSpy);

    it('should call editTodo', () => {
      editTodo();
      const spyLastCall = dispatchSpy.mock.calls[0][0];
      expect(spyLastCall.type).toEqual(dispatchSpy.mock.calls[0][0].type);
    });

    it('should call toggleTodo', () => {
      toggleTodo();
      const spyLastCall = dispatchSpy.mock.calls[0][0];
      expect(spyLastCall.type).toEqual(dispatchSpy.mock.calls[0][0].type);
    });

    it('should call deleteTodo', () => {
      deleteTodo();
      const spyLastCall = dispatchSpy.mock.calls[0][0];
      expect(spyLastCall.type).toEqual(dispatchSpy.mock.calls[0][0].type);
    });
  });
});
