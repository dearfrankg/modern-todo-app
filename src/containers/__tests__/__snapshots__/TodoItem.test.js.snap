// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Container: should render correctly with default props 1`] = `<Connect(TodoItemContainer) />`;
